pages/pipelines/buildkite_secrets.md

```diff
@@ -0,0 +1,85 @@
+
+- Setting a Buildkite secret with the key `secret_name` into an environment variable called `SECRET_VAR`:
+
+    `SECRET_VAR=$(buildkite-agent secret get secret_name)`
+
+- Redirecting the value of a Buildkite secret with the key `secret_name` into a file called `secret.txt`:
+
+    `buildkite-agent secret get secret_name > secret.txt`
+
+- Passing the output of your Buildkite secret (via the `buildkite-agent secret get` command) to your own tool named `cli-tool` that accepts a secret via its `-token` option:
+
+    `cli-tool —token $(buildkite-agent secret get secret_name)`
+
+Here’s a simple example of how one of these commands might appear in a Buildkite Pipeline step:
+
+```yaml
+steps:
+  - agents: { queue: "hosted" }
+    command:
+      - buildkite-agent secret get secret_name > secret.txt
+```
+{: codeblock-file="pipeline.yml"}
+
+## Redaction
+
+If any Pipeline, script or your own tool (accidentally) prints out the value of any Buildkite secret to standard out, this value is automatically redacted from the build logs.
+
+If for any reason you detect a secret value that isn't redacted, please rotate your secrets and contact security@buildkite.com.
+
+## Security
+
+Buildkite secrets are designed, with the following controls in place:
+
+- Secrets are encrypted in transit using TLS.
+- Secrets are always stored encrypted at rest.
+- All access to the secrets are logged.
+- Employee access to secrets is strictly limited and audited.
+
+> 📘 Buildkite secrets is not a zero-knowledge system
+> Buildkite owns, stores and manages the keys used for encrypting the secrets stored in the service at rest and in transit. Buildkite recommend customers implement additional controls to manage the lifecycle of secrets stored within Buildkite secrets, in addition to any monitoring capability you may require in line with your risk appetite.
+
+## Best practices
+
+- Do not store long lived secrets or secrets with no expiry in Buildkite secrets.
+- Only store short lived secrets in Buildkite secrets.
+- Track the secrets stored in Buildkite secrets within your own asset management processes.
+- Enable logging for services that are accessed using the secrets stored in Buildkite secrets.
+- Should you detect a compromise or are concerned about the security of your secrets, please contact security@buildkite.com immediately.
+
+## Alternative secret management methods
+
+If Buildkite secrets is not suitable for your requirements, Buildkite recommends that you manage your own secrets in your own infrastructure, and then implement [OIDC](/docs/pipelines/security/oidc) with your cloud storage/service provider to grant hosted agents access to these secrets. You can follow the [OIDC with AWS](/docs/pipelines/security/oidc/aws) process to work with secrets in AWS managed by the AWS Systems Manager (SSM). These secrets can then be accessed in your Buildkite Pipelines using hosted agents.
```

Comment:
```suggestion
If Buildkite secrets is not suitable for your requirements, Buildkite recommends that you manage your own secrets in your own infrastructure, and then implement [OIDC](/docs/pipelines/security/oidc) with your cloud storage/service provider to grant hosted agents access to these secrets. You can follow the [OIDC with AWS](/docs/pipelines/security/oidc/aws) process to work with secrets in AWS managed by the AWS Systems Manager (SSM). These secrets can then be accessed in your Buildkite Pipelines using both hosted and self-hosted agents.
```

Reply:
@moskyb , could we also change:

> to grant hosted agents access to these secrets

to:

> to grant hosted and self-hosted agents access to these secrets

?